smcontroller/smcontroller.go

```diff
@@ -340,6 +316,89 @@ func (controller *Controller) RegisterSM(stream pb.SMService_RegisterSMServer) e
  * Private
  **********************************************************************************************************************/
 
+func (controller *Controller) getSystemLog(logRequest cloudprotocol.RequestLog) error {
+	if logRequest.Filter.NodeIDs != nil {
```

Comment:
I suggest creating function: `getNodeHandlersByFilter` to get either all node handlers or filtered ones and use it in the corresponding functions.

Reply:
Done